cpp/src/arrow/ipc/reader.h

```diff
@@ -425,6 +425,37 @@ class ARROW_EXPORT StreamDecoder {
   /// \return Status
   Status Consume(std::shared_ptr<Buffer> buffer);
 
+  /// \brief Reset the internal status.
+  ///
+  /// You can reuse this decoder for new stream after calling
+  /// this. For example, you can implement endless decoder with this:
```

Comment:
Honestly, I don't think the endless decoder is very useful (why would I want an endless decoder?). Perhaps we can simply remove it?

Reply:
OK. I'll remove this example.

See #37429 for a use case of the endless decoder. It reads multiple streams with one decoder instance.

FYI: We can implement the use case by using `next_required_size()` instead but we need to recreate decoder instances manually:

```cpp
while (true) {
  auto buffer = get_data(decoder->next_required_size());
  if (!buffer) {
    break;
  }
  decoder->Consume(buffer);
  if (listener->status() == EOS) {
    decoder = create_decoder(listener);
  }
}
```